From 3a86a25de65d5b62aeee4f63245b7917962da4bc Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Fri, 11 Mar 2022 10:49:33 +0800
Subject: [PATCH] Reset the windowinsets value for wechat in secondary display

Some activity UI of wechat/xueersi display abnormally in the
secondary display windows, seems some activity layouts of wechat
have dependency on the status bar. Reset the windowinsets of the
secondary displays to incorporate the height of status bar.

Tracked-On: OAM-100827
Tracked-On: OAM-100860
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 core/java/android/view/ViewRootImpl.java | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/core/java/android/view/ViewRootImpl.java b/core/java/android/view/ViewRootImpl.java
index 18e9ef0e4d05..1ac55adb7857 100644
--- a/core/java/android/view/ViewRootImpl.java
+++ b/core/java/android/view/ViewRootImpl.java
@@ -2306,7 +2306,21 @@ public final class ViewRootImpl implements ViewParent,
             // clearing the cutout, so we don't need to dispatch the cutout to the hierarchy.
             insets = insets.consumeDisplayCutout();
         }
-        host.dispatchApplyWindowInsets(insets);
+
+        int wechatLogin = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR | View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
+        if (("com.tencent.mm".equals(mContext.getPackageName())
+                  || "com.xueersi.parentsmeeting".equals(mContext.getPackageName()))
+                  && mWindowAttributes.type == WindowManager.LayoutParams.TYPE_BASE_APPLICATION
+                  && mWindowAttributes.subtreeSystemUiVisibility != wechatLogin) {
+             WindowInsets tmpInsets = insets.replaceSystemWindowInsets(insets.getSystemWindowInsetLeft(),
+                       66,
+                       insets.getSystemWindowInsetRight(),
+                       insets.getSystemWindowInsetBottom());
+             host.dispatchApplyWindowInsets(tmpInsets);
+        } else {
+             host.dispatchApplyWindowInsets(insets);
+        }
+
         mAttachInfo.delayNotifyContentCaptureInsetsEvent(insets.getInsets(Type.all()));
         Trace.traceEnd(Trace.TRACE_TAG_VIEW);
     }
-- 
2.25.1

